src/Microsoft.OData.Core/RawValueWriter.cs

```diff
@@ -154,6 +153,86 @@ internal void Flush()
             }
         }
 
+        /// <summary>
+        /// Asynchronously start writing a raw output. This should only be called once.
+        /// </summary>
+        /// <returns>A task that represents the asynchronous write operation.</returns>
+        internal async Task StartAsync()
+        {
+            if (this.settings.HasJsonPaddingFunction())
+            {
+                await this.textWriter.WriteAsync(this.settings.JsonPCallback)
+                    .ConfigureAwait(false);
+                await this.textWriter.WriteAsync(JsonConstants.StartPaddingFunctionScope)
+                    .ConfigureAwait(false);
+            }
+        }
+
+        /// <summary>
+        /// Asynchronously end the writing of a raw output. This should be the last thing called.
+        /// </summary>
+        /// <returns>A task that represents the asynchronous write operation.</returns>
+        internal Task EndAsync()
+        {
+            if (this.settings.HasJsonPaddingFunction())
+            {
+                return this.textWriter.WriteAsync(JsonConstants.EndPaddingFunctionScope);
+            }
+
+            return TaskUtils.CompletedTask;

```

Comment:
Is this better than using `Task.CompletedTask` from the standard lib?

Reply:
Task.Completed is not supported across all frameworks that we support.